core/blockchain/block_tree.hpp

```diff
@@ -48,6 +56,28 @@ namespace kagome::blockchain {
     virtual outcome::result<primitives::Justification> getBlockJustification(
         const primitives::BlockId &block) const = 0;
 
+    /**
+     * Adds header to the storage
+     * @param header that we are adding
+     * @return result with success if header's parent exists on storage and new
+     * header was added. Error otherwise
+     */
+    virtual outcome::result<void> addBlockHeader(
+        primitives::BlockHeader header) = 0;
```

Comment:
Is it ok to add block's parts separately? Is it supported wiping added parts at fail of adding other parts?

Reply:
Substrate is designed to be "light client first". Which means that we should announce only block headers to every client. When full node receives header, it should store it and synchronize the body. 
Didn't get about wiping, it will be wiped only in case this header with block didn't become part of the finalized chain